frontend/src/metabase/visualizations/types/visualization.ts

```diff
@@ -156,18 +159,34 @@ export type ColumnSettingDefinition<TValue, TProps = unknown> = {
     col: DatasetColumn,
     settings: OptionsType,
     onChange: (value: TValue) => void,
-    extra: { series: Series },
+    extra: {
+      series: Series;
+    },
   ) => TProps;
 };
 
-export type VisualizationSettingDefinition<TValue, TProps = void> = {
+// why do we have an object of extra values when we can just use the
+// widget definition as we do for marginBottom and others?
```

Comment:
That is because we hardcode `marginBottom` in the visualization settings definitions, it does not depend on anything. But `dashboardId` and `transformedSeries` are coming from a card/dashcard